app/src/main/java/com/codelab/android/datastore/data/UserPreferencesSerializer.kt

```diff
@@ -0,0 +1,21 @@
+package com.codelab.android.datastore.data
+
+import androidx.datastore.core.CorruptionException
+import androidx.datastore.core.Serializer
+import com.codelab.android.datastore.UserPreferences
+import com.google.protobuf.InvalidProtocolBufferException
+import java.io.InputStream
+import java.io.OutputStream
+
+object UserPreferencesSerializer : Serializer<UserPreferences> {
```

Comment:
I see, protobuf serialization still not supported by @Parcelize from JetBrains? @adesamp

Reply:
yes, they have own serializer with java class generated from .proto file @mochadwi

Reply:
I see, it's required to generate the POJO yah. Have you check which task generating the POJO from Build Analyzer? @adesamp

Reply:
when we create .proto in folder src/main/proto, then we rebuild, the IDE will generated new class of java @mochadwi 

https://gitlab.com/sampingan-mobile/mobile-android-native/-/issues/269#note_489516022